app/assets/javascripts/components/materials_bin/materials_bin.js.coffee

```diff
@@ -5,30 +5,64 @@ window.MaterialsBinClass = React.createClass
     materials: React.PropTypes.array.isRequired
 
   getInitialState: ->
-    state = {
-      # selectedCategories[2] returns value of row that is selected in column 2 (or falsy value if nothing is selected).
-      # E.g. selectedCategories = [1, 3, 2] means that:
-      # - the second row is selected (idx = 1) in the first column,
-      # - the fourth row is selected (idx = 3) in the second column,
-      # - the third row is selected (idx = 2) in the third column.
-      selectedCategories: []
-    }
+    # selectedCategories[2] returns value of row that is selected in column 2 (or falsy value if nothing is selected).
+    # E.g. selectedCategories = [1, 3, 2] means that:
+    # - the second row is selected (idx = 1) in the first column,
+    # - the fourth row is selected (idx = 3) in the second column,
+    # - the third row is selected (idx = 2) in the third column.
+    selectedCategories: []
+
+  componentWillMount: ->
+    # map the category names to slugs so we have a quick lookup in #checkHash
+    @columnSlugMap = {}
+    mapSlugs = (list, column = 0) =>
+      for item, row in list
```

Comment:
Probably `row` doesn't represent a real row index here (due to format of the JSON and nested categories). It will be row of the given list, but not the whole column. In `_getColumns()` I'm calculating it in a following way:
`rowIdx = columns[columnIdx].length`.
Maybe it can be useful to reuse part of the `_getColumns` logic and work with the flat structure instead of this nested one.